application/src/main/java/org/thingsboard/server/service/ttl/cloud/CloudEventsCleanUpService.java

```diff
@@ -32,7 +33,7 @@
 @TbCoreComponent
 @Slf4j
 @Service
-@ConditionalOnExpression("${sql.ttl.cloud_events.enabled:true} && ${sql.ttl.cloud_events.cloud_events_ttl:0} > 0")
+@ConditionalOnExpression("${sql.ttl.cloud_events.enabled:true} && ${sql.ttl.cloud_events.cloud_events_ttl:0} > 0 && '${queue.type:null}' != 'kafka'")
```

Comment:
In that case, we won't clean up the leftovers from db, will we? So remove this condition and let it work if true and ttl > 0